core/src/main/java/com/dtolabs/rundeck/core/execution/workflow/steps/node/impl/DefaultScriptFileNodeStepUtils.java

```diff
@@ -184,9 +186,24 @@ public File writeScriptToTempFile(
                     expandTokens
             );
         } else if (null != serverScriptFilePath) {
-            //DON'T expand tokens in the script
-            //TODO: make token expansion optional for local file sources
-            temp = new File(serverScriptFilePath);
+            File serverScriptFile = new File(serverScriptFilePath);
+            if(expandTokens){
+                try {
+                    InputStream inputStream = new FileInputStream(serverScriptFile);
```

Comment:
use try-with-resources so the input stream is correctly closed `try(InputStream inputStream = new FileInputStream(serverScriptFile)){ ...`